certbot-apache/certbot_apache/parser.py

```diff
@@ -303,21 +303,34 @@ def get_ifmod(self, aug_conf_path, mod, beginning=False, force_create=False):
         """
         if_mods = self.aug.match(("%s/IfModule/*[self::arg='%s']" %
                                   (aug_conf_path, mod)))
-        if len(if_mods) == 0 or force_create:
-            if beginning:
-                c_path_arg = "{}/IfModule[1]/arg".format(aug_conf_path)
-                # Insert IfModule before the first directive
-                self.aug.insert("{}/directive[1]".format(aug_conf_path),
-                                "IfModule", True)
-            else:
-                c_path = "{}/IfModule[last() + 1]".format(aug_conf_path)
-                c_path_arg = "{}/IfModule[last()]/arg".format(aug_conf_path)
-                self.aug.set(c_path, "")
-            self.aug.set(c_path_arg, mod)
-            if_mods = self.aug.match(("%s/IfModule/*[self::arg='%s']" %
-                                      (aug_conf_path, mod)))
+        if not if_mods:
+            return self.create_ifmod(aug_conf_path, mod, beginning)
+
         # Strip off "arg" at end of first ifmod path
-        return if_mods[0][:len(if_mods[0]) - 3]
+        return if_mods[0].rpartition("arg")[0]
+
+    def create_ifmod(self, aug_conf_path, mod, beginning=False):
+        """Creates a new <IfMod mod> and returs it path.
+
+        :param str aug_conf_path: Augeas configuration path
+        :param str mod: module ie. mod_ssl.c
+        :param bool beginning: If the IfModule should be created to the beginning
+            of augeas path DOM tree.
+
```

Comment:
nit: can we document the return value?

Reply:
Thanks for adding this to `get_ifmod` but can you also add it to `create_ifmod`? I actually put this comment on `create_ifmod` because I didn't feel great asking you to add missing documentation to functions you didn't write in this PR, but with the change we made to `get_ifmod`'s return value, I'm glad you did so!

Reply:
Yeah, I was supposed to do that but somehow it slipped through. It's now added however.